source/particle/output/ascii.cc

```diff
@@ -106,19 +99,15 @@ namespace aspect
             output << "\n";
           }
 
-        file_index++;
+        this->increment_file_index ();
 
-        return output_file_prefix;
+        return "txt";
       }
 
       template <int dim>
       template <class Archive>
-      void ASCIIOutput<dim>::serialize (Archive &ar, const unsigned int)
-      {
-        // invoke serialization of the base class
-        ar &file_index
-        ;
-      }
+      void ASCIIOutput<dim>::serialize (Archive &, const unsigned int)
+      {}
```

Comment:
If some of the derived classes don't need to implement anything, then maybe move this function into the base class and only overload it in the derived classes if they need to do anything non-standard?